astroquery/heasarc/core.py

```diff
@@ -1,92 +1,246 @@
+
+        These are not meant to be used directly by the user.
+        """
+        if self._meta_info is None:
+            query = (
+                "SELECT split_part(name, '.', 1) AS table, "
+                "split_part(name, '.', 2) AS par, "
+                "CAST(value AS DECIMAL) AS value "
+                "FROM metainfo "
+                "WHERE (type = 'parameter' and relation = 'order') "
+                "OR relation LIKE 'defaultSearchRadius' "
+                "ORDER BY value"
+            )
+            self._meta_info = self.query_tap(query).to_table()
+            self._meta_info['value'] = np.array(
+                self._meta_info['value'], np.float32)
+            self._meta_info = self._meta_info[self._meta_info['value'] > 0]
+        return self._meta_info
+
+    def _get_default_cols(self, table_name):
+        """Get a list of default columns for a table
+
+        Parameters
+        ----------
+        table_name: str
+            The name of table as a str
+
+        Return
+        ------
+        a list of column names
 
-    def query_async(self, request_payload, *, cache=True, url=None):
         """
-        Submit a query based on a given request_payload. This allows detailed
-        control of the query to be submitted.
+        meta = self._meta[
+            (self._meta['table'] == table_name)
+            & (self._meta['par'] != '')
+        ]
+        meta.sort('value')
+        defaults = meta['par']
+        return defaults
+
+    def get_default_radius(self, table_name):
+        """Get a mission-appropriate default radius for a table
+
+        Parameters
+        ----------
+        table_name: str
+            The name of table as a str
+
+        Returns
+        -------
+        The radius as ~astropy.units
 
-        cache (bool) defaults to True. If set overrides global caching behavior.
-        See :ref:`caching documentation <astroquery_cache>`.
         """
+        meta = self._meta[
+            (self._meta['table'] == table_name)
+            & (self._meta['par'] == '')
+        ]
+        radius = np.double(meta['value'][0]) * u.arcmin
+        return radius
 
-        if url is None:
-            url = conf.server
+    def set_session(self, session):
+        """Set requests.Session to use when querying the data
 
-        response = self._request('GET', url, params=request_payload,
-                                 timeout=self.TIMEOUT, cache=cache)
-        return response
+        Parameters
+        ----------
+        session: ~requests.Session
+            The requests.Session to use
 
-    def query_mission_list(self, *, cache=True, get_query_payload=False):
         """
-        Returns a list of all available mission tables with descriptions
+        if not isinstance(session, requests.Session):
+            raise ValueError('session is not a ~requests.Session instance')
+
+        self._session = session
+
+    def tables(self, *, master=False, keywords=None):
```

Comment:
Not everything is consistent yet, but:
try to use the jargon we do already in the other vo modules. E.g. `list_catalogs` for TAP and `list_collections` for SIA. As I see one `list_tables` slipped through review, and I will listen if you insist on tables instead of catalogs, yet a verb will be needed in the name nevertheless.

Reply:
comment stands